src/panels/lovelace/editor/card-editor/hui-card-preview.ts

```diff
@@ -11,14 +18,32 @@ import { getCardElementTag } from "../../common/get-card-element-tag";
 import { createErrorCardConfig } from "../../cards/hui-error-card";
 import { computeRTL } from "../../../../common/util/compute_rtl";
 
-export class HuiCardPreview extends HTMLElement {
+export class HuiCardPreview extends LitElement {
   private _hass?: HomeAssistant;
   private _element?: LovelaceCard;
 
-  set hass(value: HomeAssistant) {
-    this._hass = value;
+  static get properties(): PropertyDeclarations {
+    return {
+      _hass: {},
+      _element: {},
+    };
+  }
+
+  protected updated(changedProperties: PropertyValues): void {
+    super.updated(changedProperties);
+    if (
```

Comment:
You should check if `rtl` value has changed, only then update it.

Reply:
Also when it changes from `rtl` to `ltr`.

Reply:
You need to add `changedProperties.has('hass')` to see if `hass` object changed. Then, look inside `hass` if the language has changed to decide if you need to compute it again.

Reply:
Honestly, there is no real reason to convert this to Lit? Why not just put this check inside `set hass`  ?

Reply:
Look at your comment from 8 days ago about doing this in updated event. Because of that I refactored it. I also thought we wanted to convert as much as possible to lit. Was trying to be inline with existing code. Let’s decide...